lib/fs.js

```diff
@@ -906,6 +906,11 @@ function writev(fd, buffers, position, callback) {
   validateBufferArray(buffers);
   callback = maybeCallback(callback || position);
 
+  if (buffers.length === 0) {
+    callback(null, 0, buffers);
```

Comment:
Does this need a nextTick?

Reply:
We currently "mix" for example the AbortSignal callbacks do not wait a nextTick and neither does `exists` if it reaches the `catch`.

Changing all of those is _probably_ a good idea wdyt? (though probably not in this PR and doing it for AbortSignal and exists likely needs CITGM and semverness?)

Reply:
Yea, I think it's quite important that callbacks are always invoked in a future tick.

Reply:
Ok, I will add it here. Feel free to open a PR for the other ones it's a simple but risky change.

Reply:
Added and also added a test for it @ronag